functions_lab.js

```diff
@@ -0,0 +1,332 @@
+// # Function Exercises
+
+// ## Question One
+
+// Write a function named double that takes in a number and returns that number times two
+
+
+// ```
+// Sample inputs and expected outputs:
+
+// input: 3, expectedOutput: 6
+// input: 0, expectedOutput: 0
+// input: -2, expectedOutput: -4
+// input: 99, expectedOutput: 198
+// ```
+// function double(num) {
+//     return num * 2
+// }
+
+const double = num => num * 2
+
+// console.log(double(5))
+// ## Question Two
+
+// Write a function named smallest that takes in two numbers and returns the smaller number
+
+// ```
+// Sample inputs and expected outputs:
+
+// inputOne: 8, inputTwo: 3, expectedOutput: 3.0
+// inputOne: 0, inputTwo: 0, expectedOutput: 0),
+// inputOne: -5, inputTwo: -3, expectedOutput: -5
+// inputOne: 2.3, inputTwo: 2.03, expectedOutput: 2.03
+// ```
+// function smallest(num1, num2) {
+    // if(num1 > num2) {
+    //     return num2
+    // } else if(num1 < num2) {
+    //     return num1
+    // } else {
+    //     return "numbers are equal"
+    // }
+// }
+const smallest = (num1, num2) => {
+    if(num1 > num2) {
+        return num2
+    } else if(num1 < num2) {
+        return num1
+    } else {
+        return "numbers are equal"
+    }
+}
+// console.log(smallest(3, 5))
+// ## Question Three
+
+// Write a function named smallestValueInArr that takes in an array of numbers and returns the smallest number
+
+
+// ```
+// Sample inputs and expected outputs:
+
+// input: [1.0,2,3,4,5,5], expectedOutput: 1
+// input: [6,5,4,3,2,1], expectedOutput: 1
+// input: [0,0,0,0,0,0,0,0,0,0], expectedOutput: 0
+// input: [-4,-59,-348,-34,-4], expectedOutput: -348
+// ```
+
+// let arr1 = [5, 7, 9, 3, 4]
+
+// function smallestValueInArr(arr) {
+//     for(let i = 0; i < arr.length; i++){
+//         let smallestArr = arr[0]
+//         if(arr[i] < smallestArr){
+//             smallestArr = arr[i]
+//             return smallestArr
+//         }
+//     }
+// }
+const smallestValueInArr = arr => {
```

Comment:
This would return as soon as you find a smaller number than the one at array[0].